x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/rule_edit.tsx

```diff
@@ -228,30 +229,50 @@ export const RuleEdit = ({
                   <></>
                 )}
                 <EuiFlexItem grow={false}>
-                  <EuiButton
-                    fill
-                    color="success"
-                    data-test-subj="saveEditedRuleButton"
-                    type="submit"
-                    iconType="check"
-                    isLoading={isSaving}
-                    onClick={async () => {
-                      setIsSaving(true);
-                      const savedRule = await onSaveRule();
-                      setIsSaving(false);
-                      if (savedRule) {
-                        onClose(RuleFlyoutCloseReason.SAVED);
-                        if (onSaveHandler) {
-                          onSaveHandler();
-                        }
-                      }
-                    }}
-                  >
-                    <FormattedMessage
-                      id="xpack.triggersActionsUI.sections.ruleEdit.saveButtonLabel"
-                      defaultMessage="Save"
-                    />
-                  </EuiButton>
+                  <EuiFlexGroup alignItems="center">
+                    {config.isUsingSecurity && (
+                      <EuiFlexItem grow={false}>
+                        <EuiIconTip
+                          type="alert"
+                          position="top"
+                          data-test-subj="changeInPrivilegesTip"
+                          content={i18n.translate(
+                            'xpack.triggersActionsUI.sections.ruleEdit.changeInPrivilegesLabel',
+                            {
+                              defaultMessage:
+                                'Saving this rule will change its privileges and might change its behavior.',
+                            }
+                          )}
+                        />
+                      </EuiFlexItem>
+                    )}
+                    <EuiFlexItem grow={false}>
+                      <EuiButton
+                        fill
+                        color="success"
+                        data-test-subj="saveEditedRuleButton"
+                        type="submit"
+                        iconType="check"
+                        isLoading={isSaving}
+                        onClick={async () => {
```

Comment:
I know this is an old code, but just out of curiosity: 
1 - Should we handle exceptions here, in case `onSaveRule` for some reason fails? Because If it fails `setIsSaving` will never be set to `false`. 
2 - Also, since we are already working on this file, wdyt about extracting these lines into a separate function? There is too much going on here and it may be easier to read if we move it into a separate function (just my humble opinion)

Reply:
Good points, I've made the changes to the feedback within this commit => https://github.com/elastic/kibana/pull/131738/commits/083dc788bd535affc713baa32bcffc85d26598c1.